edpm_ansible/ansible_plugins/modules/container_config_hash.py

```diff
@@ -145,15 +139,47 @@ def _update_container_config(self, path, config):
         os.chmod(path, 0o600)
         self.results['changed'] = True
 
-    def _get_config_hash(self, config_volume):
-        """Returns a config hash from a config_volume.
+    def _read_file(self, file):
+        """Read a given file and return its content
+        :param file: string
+        :returns: bytes
+        """
+
+        r = b''
+        with open(file, 'rb') as f:
+            while True:
+                data = f.read(BUF_SIZE)
+                r += data
+                if not data:
+                    break
+        return r
+
+    def _create_checksum(self, config_volume, exclusions=[]):
+        """Create an hash from a tar-ed version of the given
+        folder
+
+        :param config_volume: string
+        :param exclusions: list
+        :returns: string
+        """
+
+        total_hash = hashlib.new('md5')
+        for file in glob.glob(config_volume + '/**/*', recursive=True):
+            file_relpath = '/' + os.path.relpath(file, config_volume)
+            if os.path.isfile(file) and file_relpath not in exclusions:
+                total_hash.update(self._read_file(file))
+
+        h = total_hash.hexdigest()
+        return h
 
+    def _set_config_hash(self, config_volume, exclusions=[]):
+        """Create a config hash for a config_volume.
         :param config_volume: string
+        :param exclusions: list
         :returns: string
         """
-        hashfile = "%s.md5sum" % config_volume
-        if os.path.exists(hashfile):
-            return self._slurp(hashfile).strip('\n')
+        hash = self._create_checksum(config_volume, exclusions)
+        return hash
```

Comment:
The variable assignment is unnecessary here, we can just return `self._create_checksum(config_volume, exclusions)`

Reply:
ack